### PR TITLE
Move slugifier to its own package

### LINKED ISSUE #165: Put SlugifierInterface into its own repository.
We have discussed this a few times already. But I want to propose simply creating a new package for the [Slugifier interface](https://github.com/symfony-cmf/CoreBundle/tree/master/Slugifier) - `symfony-cmf/api-slugifier`.

The class would be caled `Symfony/Cmf/Api/Slugifier/SlugifierInterface`.

Maybe in the future we will find a better way, or maybe we will create more packages like this, however I think decoupling this interface from the CoreBundle is important.

### PR DESCRIPTION
**Pull request by WouterJ**
Fixes #165 

There is a new package now: https://github.com/symfony-cmf/slugifier-api

@dbu / @lsmith77 can you please at me as author to the symfony-cmf organization on packagist or add this package to packagist yourself?

/cc @dantleech

### CODE REVIEW

File: Slugifier/SlugifierInterface.php [hunk 1]
```diff
@@ -11,19 +11,12 @@
...
+
 /**
- * Slugifier interface.
- *
- * @author Daniel Leech <daniel@dantleech.com>
+ * @deprecated Since version 1.3, to be removed in 2.0. Install and use the
+ *             symfony-cmf/slugifier-api package instead.
  */
-interface SlugifierInterface
+interface SlugifierInterface extends ApiSlugifierInterface
```
Comment (WouterJ) [inline Slugifier/SlugifierInterface.php]: To be honest, this does not make sense at all: Interfaces are there to typehint, if one would typehint for the interface in the CoreBundle and get the CallbackSlugifier from the slugifier-api package, an error will appear on the screen.

In order to be BC, the interface in slugifier-api has to extend this one (the old one). However, in that case, the slugifier-api package has to require the CoreBundle and all advantages of a separate package are gone.

What should we do here?

 1. Remove everything from this bundle, documenting the BC break in UPGRADE-1.3.md and CHANGELOG.md
 1. Keep it as it is

**Comment by lsmith77** [comment]
I do not know how to grant organization rights on packagist. I added it to packagist and granted rights .. but is this really your account https://packagist.org/users/WouterJ/ ?

File: Slugifier/SlugifierInterface.php (hunk 1 above)
Comment (dantleech) [inline Slugifier/SlugifierInterface.php]: If we keep the proxy implementations in this bundle, then we are at least not breaking exisiting packages. F.e. Sulu uses the `CallbackSlugifier` and `SlugifierInterface`, if we keep these stubb implementations here then there will be no BC break, and we can move over to the new APIs when we can.

Perhaps we should just override the constructors and issue a deprecation warning?

**Comment by dbu** [comment]
we could move the core bundle interface in the slugifier-api package as well. then we could provide real BC without the need for hacks. the namespaces provided by slugifier-api would be a bit odd, but what the heck. at the same time we deprecate the old interface, of course.

this allows people to already use the new interface to be ready for 2.0.

**Comment by WouterJ** [comment]
I applied @dbu's suggestion now: https://github.com/symfony-cmf/slugifier-api/pull/1 Very hacky, but it does the trick.

@lsmith77 seems like I messed up with my account on packagist. For some reason, I don't have access to "WouterJ", but I do have access to "Wouter J".

**Comment by lsmith77** [comment]
@WouterJ ok .. switch to "Wouter J"

**Comment by lsmith77** [comment]
fine for me .. lets go?

**Comment by dbu** [comment]
we should wrap up https://github.com/symfony-cmf/slugifier-api/pull/1

if we follow my proposal, we can completely remove the interface file here and have it loaded from the api repository.

**Comment by dbu** [comment]
https://github.com/symfony-cmf/slugifier-api/issues/1 has been merged. can you rebase this one on master please? then we will also see if it bulds correctly now.

**Comment by WouterJ** [comment]
Ready to merge

**Comment by dbu** [comment]
thanks!

**Comment by dbu** [comment]
thanks. i tagged 1.3.0-RC1

**Pull request merged**